docker/lib/dependabot/docker/file_parser.rb

```diff
@@ -24,12 +24,15 @@ class FileParser < Dependabot::FileParsers::Base
 
       FROM = /FROM/i
       PLATFORM = /--platform\=(?<platform>\S+)/
-      TAG = /:(?<tag>[\w][\w.-]{0,127})/
+      TAG_NO_PREFIX = /(?<tag>[\w][\w.-]{0,127})/
+      TAG = /:#{TAG_NO_PREFIX}/
       DIGEST = /(?<digest>[0-9a-f]{64})/
       NAME = /\s+AS\s+(?<name>[\w-]+)/
       FROM_LINE =
         %r{^#{FROM}\s+(#{PLATFORM}\s+)?(#{REGISTRY}/)?
           #{IMAGE}#{TAG}?(?:@sha256:#{DIGEST})?#{NAME}?}x
+      TAG_WITH_DIGEST =
+        %r{^#{TAG_NO_PREFIX}(?:@sha256:#{DIGEST})?}x
```

Comment:
I was not expecting changes to the docker ecosystem in this PR. Can you elaborate? If it's just an unrelated change/fix, a separate PR would be greatly appreciated!